.github/workflows/main.yml

```diff
@@ -8,7 +8,8 @@ on:
 
 jobs:
   build:
-    runs-on: ubuntu-latest
+    runs-on: self-hosted
+    if: contains(["kkaempf", "jimmykarily", "andreas-kupries", "rohitsakala", "svollath", "thardeck"], "${{ github.actor }}")
```

Comment:
wouldn't this combine both #217 and #228 ? : 

```
if : contains(["kkaempf", "jimmykarily", "andreas-kupries", "rohitsakala", "svollath", "thardeck"], "${{ github.actor }}") || contains(github.event.pull_request.labels.*.name, 'safe to test')
```

https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#operators

Reply:
:thinking: 

I also don't know how much importance the `types: [ labeled ]` line has.

Reply:
I'm reading [this relatively irrelevant page](https://github.com/marketplace/actions/label-actions#examples) but afaict `types: [ labeled]` refers to the PR events that will trigger the workflow. That means the workflow it only triggered when a label is added. Combined with the label check it does what we want.

Reply:
Yes, `types: [labeled]` is the trigger. But this trigger would be valid for any actor. Which doesn't match my intention:
- if known actor, run unconditionally
- if unknown actor, run only when `safe to test` label is added

I fear this cannot be combined in a single action.

Reply:
Now I get it. You are right. I would still prefer to have them combined in one like this:

- Run PRs only when tagged. This allows us to skip builds when we are working on a PR, pushing commits but when we are not done yet. I don't mind manually tagging the PR again to trigger a build.
- Run main branch only when the actor is one of the team members (when I merge a PR, I'm the actor right?).